\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge iCNV\\ Zilu Zhou}}
\par\bigskip{\large \today}
\end{center}
\inputencoding{utf8}
\HeaderA{bambaf\_from\_vcf}{Get BAM baf information from vcf}{bambaf.Rul.from.Rul.vcf}
%
\begin{Description}\relax
If your vcf follow the format in the example, you could use this function to extract NGS baf from vcf files. Remember to load library before hands.
Save 6 lists, each list has N entry. N = \# of individuals (or vcf file)
ngs\_baf.nm: name of the bamfiles; ngs\_baf.chr: the chromosome; ngs\_baf.pos: the position of the variants;
ngs\_baf: the BAF of the variants; ngs\_baf.id: the ID of the variants; filenm:the file name
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bambaf_from_vcf(dir = ".", vcf_list, chr = NULL, projectname = "")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{dir}] The directory to all the vcf stored; default is right in this folder.

\item[\code{vcf\_list}] All the vcf names stored in vcf.list; could use command:"ls *.vcf > vcf.list" to generate.

\item[\code{chr}] Specify the chromosome you want to generate. Must be of int from 1-22. If not specify, this function will generate all chromosomes.

\item[\code{projectname}] Name of the project
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
dir='PATH/TO/FOLDER'
bambaf_from_vcf(dir,'example_vcf.list')
bambaf_from_vcf(dir,'example_vcf.list',chr=22)
load('bambaf_22.rda')
str(ngs_baf)
str(ngs_baf.pos)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{bed\_generator}{Generate BED file for WGS dataset.}{bed.Rul.generator}
%
\begin{Description}\relax
Default position generated from USCS genome browser
\end{Description}
%
\begin{Usage}
\begin{verbatim}
bed_generator(chr, hg, start = NULL, end = NULL, by = 1000)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chr}] Specify the chromosome you want to generate. Must be of int from 1-22

\item[\code{hg}] Specify the coordinate you want to generate from. Start and end position of hg19 and hg38 have been pre-implemented.

\item[\code{start}] The start position of your BED file.

\item[\code{end}] The end position of your BED file.

\item[\code{by}] The chunk of your DNA for each bin. Default 1kb.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
bed_generator(chr=22,hg=38)
bed_generator(22,38,5001,10000,by=500)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{get\_array\_intput}{Get array information from given format}{get.Rul.array.Rul.intput}
%
\begin{Description}\relax
If your array input file follow the format in the example, you could use this function to extract array LRR and baf. Remember to load library before hands.
Save 4*[\# of chr] lists, each list has N entry. N = \# of individuals
snp\_lrr: SNP LRR intensity; snp\_lrr.pos: the position of the SNPs
snp\_baf: the BAF of the SNPs; snp\_baf.pos: the position of the SNPs
\end{Description}
%
\begin{Usage}
\begin{verbatim}
get_array_intput(dir, pattern, chr = NULL, projectname = "")
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{dir}] A string. The directory path to the folder where store signal intensity file according to chr

\item[\code{pattern}] A string. The pattern of all the intensity file

\item[\code{chr}] Specify the chromosome you want to generate. Must be of int from 1-22. If not specify, this function will generate files for all chromosomes.

\item[\code{projectname}] Name of the project
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
dir='PATH/TO/FOLDER'
pattern=paste0('*.csv.arrayicnv$')
icnv_array_intput(dir,pattern,chr=22)
load('icnv_array_input_22.rda')
str(snp_lrr)
str(snp_lrr.pos)
str(snp_baf)
str(snp_baf.pos)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{iCNV\_detection}{CNV detection}{iCNV.Rul.detection}
\keyword{BAF,}{iCNV\_detection}
\keyword{CNV,}{iCNV\_detection}
\keyword{Intensity}{iCNV\_detection}
\keyword{Platform}{iCNV\_detection}
\keyword{integration,}{iCNV\_detection}
%
\begin{Description}\relax
Copy number variation detection tool for germline data. Able to combine intensity and BAF from SNP array and NGS data.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
iCNV_detection(ngs_plr = NULL, snp_lrr = NULL, ngs_baf = NULL,
  snp_baf = NULL, ngs_plr.pos = NULL, snp_lrr.pos = NULL,
  ngs_baf.pos = NULL, snp_baf.pos = NULL, maxIt = 50, visual = 0,
  projname = "iCNV", CN = 0, mu = c(-3, 0, 2), cap = FALSE)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{ngs\_plr}] A list of NGS intensity data. Each entry is an individual. If no NGS data, no need to specify.

\item[\code{snp\_lrr}] A list of SNP array intensity data. Each entry is an individual. If no SNP array data, no need to specify.

\item[\code{ngs\_baf}] A list of NGS BAF data. Each entry is an individual. If no NGS data, no need to specify.

\item[\code{snp\_baf}] A list of SNP array BAF data. Each entry is an individual. If no SNP array data, no need to specify.

\item[\code{ngs\_plr.pos}] A list of NGS intensity postion data. Each entry is an individual with dimension= (\#of bins or exons, 2(start and end position)). If no NGS data, no need to specify.

\item[\code{snp\_lrr.pos}] A list of SNP array intensity postion data. Each entry is an individual with length=\#of SNPs. If no SNP array data, no need to specify.

\item[\code{ngs\_baf.pos}] A list of NGS BAF postion data. Each entry is an individual with length=\#of BAFs. If no NGS data, no need to specify.

\item[\code{snp\_baf.pos}] A list of SNP array BAF postion data. Each entry is an individual with length=\#of BAFs. If no SNP array data, no need to specify.

\item[\code{maxIt}] An integer number indicate the maximum number of EM iteration if not converged during parameter inference. Default 50.

\item[\code{visual}] An indicator variable with value 0,1,2. 0 indicates no visualization, 1 indicates basic visualization, 2 indicates complete visualization (Note visual 2 only work for single platform and integer CN inferenced)

\item[\code{projname}] A string as the name of this project. Default 'iCNV'

\item[\code{CN}] An indicator variable with value 0,1 for whether wants to infer exact copy number. 0 no exact CN, 1 exact CN. Default 0.

\item[\code{mu}] A length tree vectur specify means of intensity in mixture normal distribution (Deletion, Diploid, Duplification). Default c(-3,0,2)

\item[\code{cap}] A boolean decides whether we cap insane intensity value due to double deletion or mutiple amplification. Default False
\end{ldescription}
\end{Arguments}
%
\begin{Value}
(1) CNV inference, contains CNV inference, Start and end position for each inference, Conditional probability for each inference, mu for mixture normal, sigma for mixture normal, probability of CNVs, Z score for each inference.

(2) exact copy number for each CNV inference, if CN=1.
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
# icnv call without genotype (just infer deletion, duplication)
projname='icnv.demo'
icnv_res0=iCNV_detection(ngs_plr,snp_lrr,
                         ngs_baf,snp_baf,
                         ngs_plr.pos,snp_lrr.pos,
                         ngs_baf.pos,snp_baf.pos,
                         projname=projname,CN=0,mu=c(-3,0,2),cap=T,visual = 1)
# icnv call with genotype inference and complete plot
projname='icnv.demo.geno'
icnv_res1=iCNV_detection(ngs_plr,snp_lrr,
                         ngs_baf,snp_baf,
                         ngs_plr.pos,snp_lrr.pos,
                         ngs_baf.pos,snp_baf.pos,
                         projname=projname,CN=1,mu=c(-3,0,2),cap=T,visual = 2)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{icnv\_output\_to\_gb}{Convert icnv.output to input for Genome Browser.}{icnv.Rul.output.Rul.to.Rul.gb}
%
\begin{Description}\relax
We could add the output to custom tracks on Genome Browser. Remeber to choose human assembly matches your input data.
We color coded the CNVs to make it as consistant as IGV. To show color, click 'User Track after submission', and edit
config to 'visibility=2 itemRgb="On"'. Color see Github page for more example.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
icnv_output_to_gb(chr, icnv.output)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{chr}] CNV chromosome

\item[\code{icnv.output}] output from output\_list\_function
\end{ldescription}
\end{Arguments}
%
\begin{Value}
matrix for Genome browser
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
icnv.output = output_list(icnv_res=icnv_res,sampleid=sampname_qc, CN=0, min_size=10000)
gb_input = icnv_output_to_gb(chr=22,icnv.output)
write.table(gb_input,file='icnv_res_gb_chr22.tab',quote=F,col.names=F,row.names=F)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{output\_list}{Generate ouput list.}{output.Rul.list}
%
\begin{Description}\relax
Generate human readable output from result calculated by iCNV\_detection function
\end{Description}
%
\begin{Usage}
\begin{verbatim}
output_list(icnv_res, sampleid = NULL, CN = 0, min_size = 0)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{sampleid}] the name of the sample, same order as the input

\item[\code{CN}] An indicator variable with value 0,1 for whether exact copy number inferred in iCNV\_detection. 0 no exact CN, 1 exact CN. Default 0.

\item[\code{min\_size}] A integer which indicate the minimum length of the CNV you are interested in. This could remove super short CNVs due to noise. Default 0. Recommend 1000.

\item[\code{testres}] CNV inference result. Output from iCNV\_detection()
\end{ldescription}
\end{Arguments}
%
\begin{Value}
output CNV list of each individual
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
icnv.output = output_list(icnv_res=icnv_res,sampleid=sampname_qc, CN=0)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plot\_intensity}{plot out the NGS plr or array lrr.}{plot.Rul.intensity}
%
\begin{Description}\relax
For quality checking purpose during intermediate steps
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plot_intensity(intensity, chr)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{intensity}] Specify the ngs\_plr object generated by CODEX or SNP array.

\item[\code{chr}] Specify the chromosome you want to generate. Must be of int from 1-22
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
plot_intensity(ngs_plr,chr)
plot_intensity(snp_lrr,chr)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plotHMMscore}{Plot CNV inference score.}{plotHMMscore}
%
\begin{Description}\relax
Plot out CNV inference score. Each row is a sample, each column is a SNP or, exon (WES) or bin (WGS). Red color indicate score
favor duplication whereas blue favor deletion.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plotHMMscore(icnv_res, h = NULL, t = NULL, subj = "score plot",
  output = NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{icnv\_res}] CNV inference result. Result from iCNV\_detection() (i.e. iCNV\_detection(...))

\item[\code{h}] start position of this plot. Default Start of the whole chromosome

\item[\code{t}] end position of this plot. Default End of the whole chromosome

\item[\code{output}] generated from output\_list\_function. If it isn't null, only CNVs in output file will be highlighted. Default NULL

\item[\code{title}] of this plot. Default "score plot"
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
icnv_res = iCNV_detection(...)
pdf(file=paste0(projname,'.pdf'),width=13,height = 10)
plotHMMscore(icnv_res,h=100000, t=200000, subj='my favorite subject')
dev.off()
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{plotindi}{Individual sample plot}{plotindi}
%
\begin{Description}\relax
Plot relationship between platforms and features for each individual. Only work for muli-platform inference.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
plotindi(ngs_plr, snp_lrr, ngs_baf, snp_baf, ngs_plr.pos, snp_lrr.pos,
  ngs_baf.pos, snp_baf.pos, icnvres, I, h = NULL, t = NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{ngs\_plr}] A list of NGS intensity data. Each entry is an individual. If no NGS data, no need to specify.

\item[\code{snp\_lrr}] A list of SNP array intensity data. Each entry is an individual. If no SNP array data, no need to specify.

\item[\code{ngs\_baf}] A list of NGS BAF data. Each entry is an individual. If no NGS data, no need to specify.

\item[\code{snp\_baf}] A list of SNP array BAF data. Each entry is an individual. If no SNP array data, no need to specify.

\item[\code{ngs\_plr.pos}] A list of NGS intensity postion data. Each entry is an individual with dimension= (\#of bins or exons, 2(start and end position)). If no NGS data, no need to specify.

\item[\code{snp\_lrr.pos}] A list of SNP array intensity postion data. Each entry is an individual with length=\#of SNPs. If no SNP array data, no need to specify.

\item[\code{ngs\_baf.pos}] A list of NGS BAF postion data. Each entry is an individual with length=\#of BAFs. If no NGS data, no need to specify.

\item[\code{snp\_baf.pos}] A list of SNP array BAF postion data. Each entry is an individual with length=\#of BAFs. If no SNP array data, no need to specify.

\item[\code{icnvres}] CNV inference result. The output from iCNV\_detection()

\item[\code{I}] Indicating the position of the individual to plot

\item[\code{h}] start position of this plot. Default Start of the whole chromosome

\item[\code{t}] end position of this plot. Default End of the whole chromosome
\end{ldescription}
\end{Arguments}
%
\begin{Value}
void
\end{Value}
%
\begin{Examples}
\begin{ExampleCode}
pdf(file=paste0(projname,'.pdf'),width=13,height = 10)
plotindi(r1L,r2L,baf1,baf2,rpos1,rpos2,bpos1,bpos2,icnvres,I,h=100000, t=200000)
dev.off()
\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
