<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta charset="utf-8"/>

<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>
<link rel="stylesheet" href="base.min.css"/>
<link rel="stylesheet" href="fancy.min.css"/>
<link rel="stylesheet" href="main.css"/>
<script src="compatibility.min.js"></script>
<script src="theViewer.min.js"></script>
<script>
try{
theViewer.defaultViewer = new theViewer.Viewer({});
}catch(e){}
</script>
<title></title>
</head>
<body>
<div id="sidebar">
<div id="outline">
</div>
</div>
<div id="page-container">
<div id="pf1" class="pf w0 h0" data-page-no="1"><div class="pc pc1 w0 h0"><img class="bi x0 y0 w1 h1" alt="" src="bg1.png"/><div class="t m0 x1 h2 y1 ff1 fs0 fc0 sc0 ls0 ws0">iCNV</div><div class="t m0 x2 h3 y2 ff2 fs1 fc0 sc0 ls0 ws0">December 13, 2017</div><div class="t m0 x3 h4 y3 ff3 fs2 fc0 sc0 ls0 ws0">bambaf_from_vcf<span class="_ _0"> </span><span class="ff4">Get BAM baf information fr<span class="_ _1"></span>om vcf</span></div><div class="t m0 x0 h5 y4 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y5 ff2 fs2 fc0 sc0 ls0 ws0">If<span class="_ _2"> </span>your<span class="_ _2"> </span>vcf<span class="_ _2"> </span>follow<span class="_ _2"> </span>the<span class="_ _2"> </span>format<span class="_ _2"> </span>in<span class="_ _2"> </span>the<span class="_ _2"> </span>e<span class="_ _3"></span>xample,<span class="_ _2"> </span>you<span class="_ _2"> </span>could<span class="_ _2"> </span>use<span class="_ _2"> </span>this<span class="_ _2"> </span>function<span class="_ _2"> </span>to<span class="_ _2"> </span>extract<span class="_ _2"> </span>NGS<span class="_ _2"> </span>baf<span class="_ _2"> </span>from<span class="_ _2"> </span>vcf</div><div class="t m0 x4 h6 y6 ff2 fs2 fc0 sc0 ls0 ws0">ﬁles.<span class="_ _4"> </span>Remember<span class="_ _2"> </span>to<span class="_ _2"> </span>load library<span class="_ _2"> </span>before<span class="_ _2"> </span>hands.<span class="_ _4"> </span>Sa<span class="_ _3"></span>ve<span class="_ _2"> </span>6<span class="_ _2"> </span>lists, each<span class="_ _2"> </span>list<span class="_ _2"> </span>has<span class="_ _2"> </span>N<span class="_ _2"> </span>entry<span class="_ _1"></span>.<span class="_ _4"> </span>N =<span class="_ _2"> </span>#<span class="_ _2"> </span>of<span class="_ _2"> </span>individuals</div><div class="t m0 x4 h6 y7 ff2 fs2 fc0 sc0 ls0 ws0">(or<span class="_ _5"> </span>vcf<span class="_ _5"> </span>ﬁle)<span class="_ _5"> </span>ngs_baf.nm:<span class="_ _6"> </span>name<span class="_ _5"> </span>of<span class="_ _5"> </span>the<span class="_ _5"> </span>bamﬁles;<span class="_ _7"> </span>ngs_baf.chr:<span class="_ _6"> </span>the<span class="_ _5"> </span>chromosome;<span class="_ _7"> </span>ngs_baf.pos:<span class="_ _6"> </span>the</div><div class="t m0 x4 h6 y8 ff2 fs2 fc0 sc0 ls0 ws0">position<span class="_ _7"> </span>of<span class="_ _7"> </span>the<span class="_ _8"> </span>variants;<span class="_ _9"> </span>ngs_baf:<span class="_ _a"> </span>the<span class="_ _7"> </span>BAF<span class="_ _7"> </span>of<span class="_ _7"> </span>the<span class="_ _7"> </span>variants;<span class="_ _9"> </span>ngs_baf.id:<span class="_ _a"> </span>the<span class="_ _7"> </span>ID<span class="_ _7"> </span>of<span class="_ _8"> </span>the<span class="_ _7"> </span>variants;</div><div class="t m0 x4 h6 y9 ff2 fs2 fc0 sc0 ls0 ws0">ﬁlenm:the ﬁle name</div><div class="t m0 x0 h5 ya ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 yb ff3 fs2 fc0 sc0 ls0 ws0">bambaf_from_vcf(dir<span class="_ _6"> </span>=<span class="_ _6"> </span>&quot;.&quot;,<span class="_ _6"> </span>vcf_list,<span class="_ _6"> </span>chr<span class="_ _6"> </span>=<span class="_ _b"> </span>NULL,<span class="_ _6"> </span>projectname<span class="_ _6"> </span>=<span class="_ _6"> </span>&quot;&quot;)</div><div class="t m0 x0 h5 yc ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 yd ff3 fs2 fc0 sc0 ls0 ws0">dir<span class="_ _c"> </span><span class="ff2">The directory to all the vcf stored; default is right in this folder<span class="_ _1"></span>.</span></div><div class="t m0 x4 h6 ye ff3 fs2 fc0 sc0 ls0 ws0">vcf_list<span class="_ _d"> </span><span class="ff2">All<span class="_ _4"> </span>the<span class="_ _4"> </span>vcf<span class="_ _4"> </span>names<span class="_ _4"> </span>stored<span class="_ _4"> </span>in vcf.list;<span class="_ _5"> </span>could<span class="_ _4"> </span>use<span class="_ _4"> </span>command:&quot;ls<span class="_ _4"> </span>*.vcf<span class="_ _4"> </span>&gt; vcf.list&quot;<span class="_ _4"> </span>to</span></div><div class="t m0 x5 h6 yf ff2 fs2 fc0 sc0 ls0 ws0">generate.</div><div class="t m0 x4 h6 y10 ff3 fs2 fc0 sc0 ls0 ws0">chr<span class="_ _c"> </span><span class="ff2">Specify the chromosome you want to generate.<span class="_ _4"> </span>Must be of int from 1-22.<span class="_ _5"> </span>If not</span></div><div class="t m0 x5 h6 y11 ff2 fs2 fc0 sc0 ls0 ws0">specify<span class="_ _1"></span>, this function will generate all chromosomes.</div><div class="t m0 x4 h6 y12 ff3 fs2 fc0 sc0 ls0 ws0">projectname<span class="_ _e"> </span><span class="ff2">Name of the project</span></div><div class="t m0 x0 h5 y13 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1"></span>alue</div><div class="t m0 x4 h6 y14 ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 y15 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 y16 ff3 fs3 fc0 sc0 ls0 ws0">dir=<span class="ff5">&apos;</span>PATH/TO/FOLDER<span class="ff5">&apos;</span></div><div class="t m0 x4 h8 y17 ff3 fs3 fc0 sc0 ls0 ws0">bambaf_from_vcf(dir,<span class="ff5">&apos;</span>example_vcf.list<span class="ff5">&apos;</span>)</div><div class="t m0 x4 h8 y18 ff3 fs3 fc0 sc0 ls0 ws0">bambaf_from_vcf(dir,<span class="ff5">&apos;</span>example_vcf.list<span class="ff5">&apos;</span>,chr=22)</div><div class="t m0 x4 h8 y19 ff3 fs3 fc0 sc0 ls0 ws0">load(<span class="ff5">&apos;</span>bambaf_22.rda<span class="ff5">&apos;</span>)</div><div class="t m0 x4 h8 y1a ff3 fs3 fc0 sc0 ls0 ws0">str(ngs_baf)</div><div class="t m0 x4 h8 y1b ff3 fs3 fc0 sc0 ls0 ws0">str(ngs_baf.pos)</div><div class="t m0 x6 h6 y1c ff2 fs2 fc0 sc0 ls0 ws0">1</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf2" class="pf w0 h0" data-page-no="2"><div class="pc pc2 w0 h0"><img class="bi x0 y1d w1 h9" alt="" src="bg2.png"/><div class="t m0 x0 h6 y1e ff2 fs2 fc0 sc0 ls0 ws0">2<span class="_ _f"> </span><span class="ff6">get_array_intput</span></div><div class="t m0 x3 h4 y1f ff3 fs2 fc0 sc0 ls0 ws0">bed_generator<span class="_ _10"> </span><span class="ff4">Generate BED ﬁle for WGS dataset.</span></div><div class="t m0 x0 h5 y20 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y21 ff2 fs2 fc0 sc0 ls0 ws0">Default position generated from USCS genome bro<span class="_ _3"></span>wser</div><div class="t m0 x0 h5 y22 ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 y23 ff3 fs2 fc0 sc0 ls0 ws0">bed_generator(chr,<span class="_ _6"> </span>hg,<span class="_ _6"> </span>start<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>end<span class="_ _b"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>by<span class="_ _6"> </span>=<span class="_ _6"> </span>1000)</div><div class="t m0 x0 h5 y24 ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 y25 ff3 fs2 fc0 sc0 ls0 ws0">chr<span class="_ _c"> </span><span class="ff2">Specify the chromosome you want to generate.<span class="_ _4"> </span>Must be of int from 1-22</span></div><div class="t m0 x4 h6 y26 ff3 fs2 fc0 sc0 ls0 ws0">hg<span class="_ _11"> </span><span class="ff2">Specify<span class="_ _5"> </span>the<span class="_ _7"> </span>coordinate<span class="_ _5"> </span>you<span class="_ _7"> </span>want<span class="_ _5"> </span>to<span class="_ _7"> </span>generate<span class="_ _5"> </span>from.<span class="_ _12"> </span>Start<span class="_ _5"> </span>and<span class="_ _7"> </span>end<span class="_ _5"> </span>position<span class="_ _7"> </span>of</span></div><div class="t m0 x5 h6 y27 ff2 fs2 fc0 sc0 ls0 ws0">hg19 and hg38 hav<span class="_ _3"></span>e been pre-implemented.</div><div class="t m0 x4 h6 y28 ff3 fs2 fc0 sc0 ls0 ws0">start<span class="_ _13"> </span><span class="ff2">The start position of your BED ﬁle.</span></div><div class="t m0 x4 h6 y29 ff3 fs2 fc0 sc0 ls0 ws0">end<span class="_ _c"> </span><span class="ff2">The end position of your BED ﬁle.</span></div><div class="t m0 x4 h6 y2a ff3 fs2 fc0 sc0 ls0 ws0">by<span class="_ _11"> </span><span class="ff2">The chunk of your DN<span class="_ _3"></span>A for each bin.<span class="_ _4"> </span>Default 1kb<span class="_ _3"></span>.</span></div><div class="t m0 x0 h5 y2b ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1"></span>alue</div><div class="t m0 x4 h6 y2c ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 y2d ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 y2e ff3 fs3 fc0 sc0 ls0 ws0">bed_generator(chr=22,hg=38)</div><div class="t m0 x4 h8 y2f ff3 fs3 fc0 sc0 ls0 ws0">bed_generator(22,38,5001,10000,by=500)</div><div class="t m0 x3 h4 y30 ff3 fs2 fc0 sc0 ls0 ws0">get_array_intput<span class="_ _14"> </span><span class="ff4">Get array information fr<span class="_ _1"></span>om given format</span></div><div class="t m0 x0 h5 y31 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y32 ff2 fs2 fc0 sc0 ls0 ws0">If your<span class="_ _2"> </span>array input<span class="_ _2"> </span>ﬁle follow<span class="_ _2"> </span>the format<span class="_ _2"> </span>in the e<span class="_ _3"></span>xample, you<span class="_ _2"> </span>could use this<span class="_ _2"> </span>function to e<span class="_ _3"></span>xtract array</div><div class="t m0 x4 h6 y33 ff2 fs2 fc0 sc0 ls0 ws0">LRR and baf.<span class="_ _4"> </span>Remember to load library<span class="_ _2"> </span>before hands.<span class="_ _5"> </span>Sa<span class="_ _3"></span>ve 4*[#<span class="_ _2"> </span>of chr] lists, each list has N entry<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y34 ff2 fs2 fc0 sc0 ls0 ws0">N<span class="_ _4"> </span>=<span class="_ _5"> </span>#<span class="_ _4"> </span>of<span class="_ _5"> </span>indi<span class="_ _1"></span>viduals<span class="_ _5"> </span>snp_lrr:<span class="_ _8"> </span>SNP<span class="_ _5"> </span>LRR<span class="_ _4"> </span>intensity;<span class="_ _5"> </span>snp_lrr<span class="_ _3"></span>.pos:<span class="_ _8"> </span>the<span class="_ _5"> </span>position<span class="_ _4"> </span>of<span class="_ _4"> </span>the<span class="_ _5"> </span>SNPs<span class="_ _4"> </span>snp_baf:</div><div class="t m0 x4 h6 y35 ff2 fs2 fc0 sc0 ls0 ws0">the B<span class="_ _3"></span>AF of the SNPs; snp_baf.pos:<span class="_ _4"> </span>the position of the SNPs</div><div class="t m0 x0 h5 y36 ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 y37 ff3 fs2 fc0 sc0 ls0 ws0">get_array_intput(dir,<span class="_ _6"> </span>pattern,<span class="_ _6"> </span>chr<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>projectname<span class="_ _b"> </span>=<span class="_ _6"> </span>&quot;&quot;)</div><div class="t m0 x0 h5 y38 ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 y39 ff3 fs2 fc0 sc0 ls0 ws0">dir<span class="_ _c"> </span><span class="ff2">A<span class="_ _5"> </span>string.<span class="_ _b"> </span>The<span class="_ _5"> </span>directory<span class="_ _5"> </span>path<span class="_ _5"> </span>to<span class="_ _5"> </span>the<span class="_ _4"> </span>folder<span class="_ _5"> </span>where<span class="_ _5"> </span>store<span class="_ _5"> </span>signal<span class="_ _4"> </span>intensity<span class="_ _5"> </span>ﬁle<span class="_ _5"> </span>ac-</span></div><div class="t m0 x5 h6 y3a ff2 fs2 fc0 sc0 ls0 ws0">cording to chr</div><div class="t m0 x4 h6 y3b ff3 fs2 fc0 sc0 ls0 ws0">pattern<span class="_ _15"> </span><span class="ff2">A string.<span class="_ _4"> </span>The pattern of all the intensity ﬁle</span></div><div class="t m0 x4 h6 y3c ff3 fs2 fc0 sc0 ls0 ws0">chr<span class="_ _c"> </span><span class="ff2">Specify the chromosome you want to generate.<span class="_ _4"> </span>Must be of int from 1-22.<span class="_ _5"> </span>If not</span></div><div class="t m0 x5 h6 y3d ff2 fs2 fc0 sc0 ls0 ws0">specify<span class="_ _1"></span>, this function will generate ﬁles for all chromosomes.</div><div class="t m0 x4 h6 y3e ff3 fs2 fc0 sc0 ls0 ws0">projectname<span class="_ _e"> </span><span class="ff2">Name of the project</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf3" class="pf w0 h0" data-page-no="3"><div class="pc pc3 w0 h0"><img class="bi x0 y3f w1 ha" alt="" src="bg3.png"/><div class="t m0 x0 h6 y1e ff6 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="_ _16"> </span><span class="ff2">3</span></div><div class="t m0 x0 h5 y40 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1"></span>alue</div><div class="t m0 x4 h6 y41 ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 y42 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 y43 ff3 fs3 fc0 sc0 ls0 ws0">dir=<span class="ff5">&apos;</span>PATH/TO/FOLDER<span class="ff5">&apos;</span></div><div class="t m0 x4 h8 y44 ff3 fs3 fc0 sc0 ls0 ws0">pattern=paste0(<span class="ff5">&apos;</span>*.csv.arrayicnv$<span class="ff5">&apos;</span>)</div><div class="t m0 x4 h8 y45 ff3 fs3 fc0 sc0 ls0 ws0">icnv_array_intput(dir,pattern,chr=22)</div><div class="t m0 x4 h8 y46 ff3 fs3 fc0 sc0 ls0 ws0">load(<span class="ff5">&apos;</span>icnv_array_input_22.rda<span class="ff5">&apos;</span>)</div><div class="t m0 x4 h8 y47 ff3 fs3 fc0 sc0 ls0 ws0">str(snp_lrr)</div><div class="t m0 x4 h8 y48 ff3 fs3 fc0 sc0 ls0 ws0">str(snp_lrr.pos)</div><div class="t m0 x4 h8 y49 ff3 fs3 fc0 sc0 ls0 ws0">str(snp_baf)</div><div class="t m0 x4 h8 y4a ff3 fs3 fc0 sc0 ls0 ws0">str(snp_baf.pos)</div><div class="t m0 x3 h4 y4b ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="_ _17"> </span><span class="ff4">CNV detection</span></div><div class="t m0 x0 h5 y4c ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y4d ff2 fs2 fc0 sc0 ls0 ws0">Copy number<span class="_ _4"> </span>variation detection<span class="_ _4"> </span>tool for<span class="_ _4"> </span>germline data.<span class="_ _8"> </span>Able<span class="_ _4"> </span>to combine<span class="_ _4"> </span>intensity and<span class="_ _4"> </span>B<span class="_ _3"></span>AF<span class="_ _4"> </span>from</div><div class="t m0 x4 h6 y4e ff2 fs2 fc0 sc0 ls0 ws0">SNP array and NGS data.</div><div class="t m0 x0 h5 y4f ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 y50 ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection(ngs_plr<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>snp_lrr<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _b"> </span>ngs_baf<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,</div><div class="t m0 x7 h7 y51 ff3 fs2 fc0 sc0 ls0 ws0">snp_baf<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>ngs_plr.pos<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _b"> </span>snp_lrr.pos<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,</div><div class="t m0 x7 h7 y52 ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf.pos<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>snp_baf.pos<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _b"> </span>maxIt<span class="_ _6"> </span>=<span class="_ _6"> </span>50,<span class="_ _6"> </span>visual<span class="_ _6"> </span>=<span class="_ _6"> </span>0,</div><div class="t m0 x7 h7 y53 ff3 fs2 fc0 sc0 ls0 ws0">projname<span class="_ _6"> </span>=<span class="_ _6"> </span>&quot;iCNV&quot;,<span class="_ _6"> </span>CN<span class="_ _6"> </span>=<span class="_ _6"> </span>0,<span class="_ _b"> </span>mu<span class="_ _6"> </span>=<span class="_ _6"> </span>c(-3,<span class="_ _6"> </span>0,<span class="_ _6"> </span>2),<span class="_ _6"> </span>cap<span class="_ _b"> </span>=<span class="_ _6"> </span>FALSE)</div><div class="t m0 x0 h5 y54 ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 y55 ff3 fs2 fc0 sc0 ls0 ws0">ngs_plr<span class="_ _15"> </span><span class="ff2">A<span class="_ _4"> </span>list<span class="_ _5"> </span>of<span class="_ _4"> </span>NGS<span class="_ _5"> </span>intensity<span class="_ _4"> </span>data.<span class="_ _6"> </span>Each<span class="_ _5"> </span>entry<span class="_ _4"> </span>is<span class="_ _4"> </span>an<span class="_ _5"> </span>indi<span class="_ _3"></span>vidual.<span class="_ _6"> </span>If<span class="_ _5"> </span>no<span class="_ _4"> </span>NGS<span class="_ _4"> </span>data,<span class="_ _5"> </span>no</span></div><div class="t m0 x5 h6 y56 ff2 fs2 fc0 sc0 ls0 ws0">need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y57 ff3 fs2 fc0 sc0 ls0 ws0">snp_lrr<span class="_ _15"> </span><span class="ff2">A list<span class="_ _4"> </span>of SNP<span class="_ _4"> </span>array intensity<span class="_ _4"> </span>data.<span class="_ _5"> </span>Each<span class="_ _4"> </span>entry is<span class="_ _4"> </span>an individual.<span class="_ _5"> </span>If<span class="_ _4"> </span>no SNP array</span></div><div class="t m0 x5 h6 y58 ff2 fs2 fc0 sc0 ls0 ws0">data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y59 ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf<span class="_ _15"> </span><span class="ff2">A<span class="_ _4"> </span>list of<span class="_ _4"> </span>NGS<span class="_ _4"> </span>B<span class="_ _1"></span>AF<span class="_ _4"> </span>data.<span class="_ _8"> </span>Each<span class="_ _4"> </span>entry is<span class="_ _4"> </span>an<span class="_ _4"> </span>indi<span class="_ _3"></span>vidual.<span class="_ _7"> </span>If<span class="_ _4"> </span>no<span class="_ _4"> </span>NGS data,<span class="_ _4"> </span>no<span class="_ _4"> </span>need</span></div><div class="t m0 x5 h6 y5a ff2 fs2 fc0 sc0 ls0 ws0">to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y5b ff3 fs2 fc0 sc0 ls0 ws0">snp_baf<span class="_ _15"> </span><span class="ff2">A list<span class="_ _2"> </span>of SNP array<span class="_ _2"> </span>BAF<span class="_ _2"> </span>data.<span class="_ _4"> </span>Each entry is an<span class="_ _2"> </span>individual.<span class="_ _4"> </span>If no<span class="_ _2"> </span>SNP array data,</span></div><div class="t m0 x5 h6 y5c ff2 fs2 fc0 sc0 ls0 ws0">no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y5d ff3 fs2 fc0 sc0 ls0 ws0">ngs_plr.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _5"> </span>list<span class="_ _4"> </span>of<span class="_ _5"> </span>NGS<span class="_ _4"> </span>intensity<span class="_ _5"> </span>postion<span class="_ _4"> </span>data.<span class="_ _b"> </span>Each<span class="_ _5"> </span>entry<span class="_ _5"> </span>is<span class="_ _4"> </span>an<span class="_ _5"> </span>indi<span class="_ _3"></span>vidual<span class="_ _5"> </span>with<span class="_ _4"> </span>dimen-</span></div><div class="t m0 x5 h6 y5e ff2 fs2 fc0 sc0 ls0 ws0">sion= (#of bins<span class="_ _4"> </span>or exons, 2(start and end<span class="_ _4"> </span>position)).<span class="_ _5"> </span>If no NGS data,<span class="_ _4"> </span>no need to</div><div class="t m0 x5 h6 y5f ff2 fs2 fc0 sc0 ls0 ws0">specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y60 ff3 fs2 fc0 sc0 ls0 ws0">snp_lrr.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _7"> </span>list<span class="_ _7"> </span>of<span class="_ _7"> </span>SNP<span class="_ _7"> </span>array<span class="_ _7"> </span>intensity<span class="_ _7"> </span>postion<span class="_ _5"> </span>data.<span class="_ _18"> </span>Each<span class="_ _7"> </span>entry<span class="_ _7"> </span>is<span class="_ _7"> </span>an<span class="_ _7"> </span>indi<span class="_ _3"></span>vidual<span class="_ _7"> </span>with</span></div><div class="t m0 x5 h6 y61 ff2 fs2 fc0 sc0 ls0 ws0">length=#of SNPs.<span class="_ _4"> </span>If no SNP array data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y62 ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _4"> </span>list<span class="_ _5"> </span>of<span class="_ _5"> </span>NGS<span class="_ _4"> </span>B<span class="_ _3"></span>AF<span class="_ _5"> </span>postion<span class="_ _4"> </span>data.<span class="_ _b"> </span>Each<span class="_ _5"> </span>entry<span class="_ _4"> </span>is<span class="_ _5"> </span>an<span class="_ _4"> </span>individual<span class="_ _4"> </span>with<span class="_ _5"> </span>length=#of</span></div><div class="t m0 x5 h6 y63 ff2 fs2 fc0 sc0 ls0 ws0">B<span class="_ _3"></span>AFs.<span class="_ _4"> </span>If no NGS data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 y64 ff3 fs2 fc0 sc0 ls0 ws0">snp_baf.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _2"> </span>list<span class="_ _2"> </span>of<span class="_ _19"> </span>SNP<span class="_ _19"> </span>array<span class="_ _2"> </span>BAF<span class="_ _19"> </span>postion<span class="_ _19"> </span>data.<span class="_ _4"> </span>Each<span class="_ _2"> </span>entry<span class="_ _19"> </span>is<span class="_ _2"> </span>an<span class="_ _2"> </span>individual<span class="_ _19"> </span>with<span class="_ _19"> </span>length=#of</span></div><div class="t m0 x5 h6 y3e ff2 fs2 fc0 sc0 ls0 ws0">B<span class="_ _3"></span>AFs.<span class="_ _4"> </span>If no SNP array data, no need to specify<span class="_ _1"></span>.</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf4" class="pf w0 h0" data-page-no="4"><div class="pc pc4 w0 h0"><img class="bi x0 y65 w1 h1" alt="" src="bg4.png"/><div class="t m0 x0 h6 y1e ff2 fs2 fc0 sc0 ls0 ws0">4<span class="_ _1a"> </span><span class="ff6">icn<span class="_ _3"></span>v_output_to_gb</span></div><div class="t m0 x4 h6 y40 ff3 fs2 fc0 sc0 ls0 ws0">maxIt<span class="_ _13"> </span><span class="ff2">An<span class="_ _5"> </span>inte<span class="_ _3"></span>ger<span class="_ _5"> </span>number<span class="_ _4"> </span>indicate<span class="_ _5"> </span>the<span class="_ _5"> </span>maximum<span class="_ _4"> </span>number<span class="_ _5"> </span>of<span class="_ _5"> </span>EM<span class="_ _4"> </span>iteration<span class="_ _5"> </span>if<span class="_ _5"> </span>not<span class="_ _4"> </span>con-</span></div><div class="t m0 x5 h6 y66 ff2 fs2 fc0 sc0 ls0 ws0">ver<span class="_ _3"></span>ged during parameter inference.<span class="_ _4"> </span>Default 50.</div><div class="t m0 x4 h6 y67 ff3 fs2 fc0 sc0 ls0 ws0">visual<span class="_ _1b"> </span><span class="ff2">An<span class="_ _4"> </span>indicator variable with<span class="_ _4"> </span>v<span class="_ _3"></span>alue<span class="_ _4"> </span>0,1,2.<span class="_ _7"> </span>0 indicates<span class="_ _4"> </span>no<span class="_ _4"> </span>visualization, 1<span class="_ _4"> </span>indicates</span></div><div class="t m0 x5 h6 y68 ff2 fs2 fc0 sc0 ls0 ws0">basic visualization, 2 indicates complete<span class="_ _4"> </span>visualization (Note visual 2 only work</div><div class="t m0 x5 h6 y69 ff2 fs2 fc0 sc0 ls0 ws0">for single platform and integer CN inferenced)</div><div class="t m0 x4 h6 y6a ff3 fs2 fc0 sc0 ls0 ws0">projname<span class="_ _d"> </span><span class="ff2">A string as the name of this project.<span class="_ _4"> </span>Default ’iCNV’</span></div><div class="t m0 x4 h6 y6b ff3 fs2 fc0 sc0 ls0 ws0">CN<span class="_ _11"> </span><span class="ff2">An indicator v<span class="_ _3"></span>ariable with v<span class="_ _1"></span>alue 0,1 for whether wants<span class="_ _2"> </span>to infer exact cop<span class="_ _3"></span>y num-</span></div><div class="t m0 x5 h6 y6c ff2 fs2 fc0 sc0 ls0 ws0">ber<span class="_ _1"></span>.<span class="_ _5"> </span>0 no e<span class="_ _3"></span>xact CN, 1 exact CN. Default 0.</div><div class="t m0 x4 h6 y6d ff3 fs2 fc0 sc0 ls0 ws0">mu<span class="_ _11"> </span><span class="ff2">A<span class="_ _5"> </span>length<span class="_ _4"> </span>tree<span class="_ _5"> </span>v<span class="_ _3"></span>ectur<span class="_ _5"> </span>specify<span class="_ _4"> </span>means<span class="_ _5"> </span>of<span class="_ _4"> </span>intensity<span class="_ _5"> </span>in<span class="_ _5"> </span>mixture<span class="_ _4"> </span>normal<span class="_ _5"> </span>distrib<span class="_ _3"></span>ution</span></div><div class="t m0 x5 h6 y6e ff2 fs2 fc0 sc0 ls0 ws0">(Deletion, Diploid, Dupliﬁcation).<span class="_ _4"> </span>Default c(-3,0,2)</div><div class="t m0 x4 h6 y6f ff3 fs2 fc0 sc0 ls0 ws0">cap<span class="_ _c"> </span><span class="ff2">A boolean<span class="_ _2"> </span>decides whether we cap<span class="_ _2"> </span>insane intensity v<span class="_ _3"></span>alue due<span class="_ _2"> </span>to double deletion</span></div><div class="t m0 x5 h6 y70 ff2 fs2 fc0 sc0 ls0 ws0">or mutiple ampliﬁcation.<span class="_ _4"> </span>Default False</div><div class="t m0 x0 h5 y71 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1"></span>alue</div><div class="t m0 x4 h6 y72 ff2 fs2 fc0 sc0 ls0 ws0">(1) CNV inference, contains CNV<span class="_ _2"> </span>inference, Start and end position for each<span class="_ _2"> </span>inference, Conditional</div><div class="t m0 x4 h6 y73 ff2 fs2 fc0 sc0 ls0 ws0">probability<span class="_ _5"> </span>for<span class="_ _5"> </span>each<span class="_ _5"> </span>inference,<span class="_ _7"> </span>mu<span class="_ _5"> </span>for<span class="_ _5"> </span>mixture<span class="_ _5"> </span>normal,<span class="_ _5"> </span>sigma<span class="_ _5"> </span>for<span class="_ _5"> </span>mixture<span class="_ _5"> </span>normal,<span class="_ _7"> </span>probability<span class="_ _5"> </span>of</div><div class="t m0 x4 h6 y74 ff2 fs2 fc0 sc0 ls0 ws0">CNVs, Z score for each inference.</div><div class="t m0 x4 h6 y75 ff2 fs2 fc0 sc0 ls0 ws0">(2) exact copy number for each CNV inference, if CN=1.</div><div class="t m0 x0 h5 y76 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 y77 ff3 fs3 fc0 sc0 ls0 ws0">#<span class="_ _9"> </span>icnv<span class="_ _9"> </span>call<span class="_ _9"> </span>without<span class="_ _9"> </span>genotype<span class="_ _9"> </span>(just<span class="_ _9"> </span>infer<span class="_ _9"> </span>deletion,<span class="_ _9"> </span>duplication)</div><div class="t m0 x4 h8 y78 ff3 fs3 fc0 sc0 ls0 ws0">projname=<span class="ff5">&apos;</span>icnv.demo<span class="ff5">&apos;</span></div><div class="t m0 x4 h8 y79 ff3 fs3 fc0 sc0 ls0 ws0">icnv_res0=iCNV_detection(ngs_plr,snp_lrr,</div><div class="t m0 x8 h8 y7a ff3 fs3 fc0 sc0 ls0 ws0">ngs_baf,snp_baf,</div><div class="t m0 x8 h8 y7b ff3 fs3 fc0 sc0 ls0 ws0">ngs_plr.pos,snp_lrr.pos,</div><div class="t m0 x8 h8 y7c ff3 fs3 fc0 sc0 ls0 ws0">ngs_baf.pos,snp_baf.pos,</div><div class="t m0 x8 h8 y7d ff3 fs3 fc0 sc0 ls0 ws0">projname=projname,CN=0,mu=c(-3,0,2),cap=T,visual<span class="_ _9"> </span>=<span class="_ _9"> </span>1)</div><div class="t m0 x4 h8 y7e ff3 fs3 fc0 sc0 ls0 ws0">#<span class="_ _9"> </span>icnv<span class="_ _9"> </span>call<span class="_ _9"> </span>with<span class="_ _9"> </span>genotype<span class="_ _9"> </span>inference<span class="_ _9"> </span>and<span class="_ _9"> </span>complete<span class="_ _9"> </span>plot</div><div class="t m0 x4 h8 y7f ff3 fs3 fc0 sc0 ls0 ws0">projname=<span class="ff5">&apos;</span>icnv.demo.geno<span class="ff5">&apos;</span></div><div class="t m0 x4 h8 y80 ff3 fs3 fc0 sc0 ls0 ws0">icnv_res1=iCNV_detection(ngs_plr,snp_lrr,</div><div class="t m0 x8 h8 y81 ff3 fs3 fc0 sc0 ls0 ws0">ngs_baf,snp_baf,</div><div class="t m0 x8 h8 y82 ff3 fs3 fc0 sc0 ls0 ws0">ngs_plr.pos,snp_lrr.pos,</div><div class="t m0 x8 h8 y83 ff3 fs3 fc0 sc0 ls0 ws0">ngs_baf.pos,snp_baf.pos,</div><div class="t m0 x8 h8 y84 ff3 fs3 fc0 sc0 ls0 ws0">projname=projname,CN=1,mu=c(-3,0,2),cap=T,visual<span class="_ _9"> </span>=<span class="_ _9"> </span>2)</div><div class="t m0 x3 h4 y85 ff3 fs2 fc0 sc0 ls0 ws0">icnv_output_to_gb<span class="_ _1c"> </span><span class="ff4">Con<span class="_ _3"></span>vert icn<span class="_ _3"></span>v<span class="_ _1"></span>.output to input for Genome Br<span class="_ _1"></span>owser<span class="_ _1"></span>.</span></div><div class="t m0 x0 h5 y86 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y87 ff2 fs2 fc0 sc0 ls0 ws0">W<span class="_ _1"></span>e<span class="_ _2"> </span>could<span class="_ _2"> </span>add<span class="_ _2"> </span>the<span class="_ _2"> </span>output<span class="_ _2"> </span>to<span class="_ _2"> </span>custom<span class="_ _2"> </span>tracks<span class="_ _2"> </span>on<span class="_ _2"> </span>Genome Bro<span class="_ _1"></span>wser<span class="_ _3"></span>.<span class="_ _4"> </span>Remeber<span class="_ _2"> </span>to<span class="_ _2"> </span>choose<span class="_ _2"> </span>human<span class="_ _2"> </span>assembly</div><div class="t m0 x4 h6 y88 ff2 fs2 fc0 sc0 ls0 ws0">matches your input data.<span class="_ _4"> </span>W<span class="_ _1"></span>e color coded the CNVs to make it<span class="_ _2"> </span>as consistant as IGV<span class="_ _1d"></span>. T<span class="_ _1"></span>o sho<span class="_ _3"></span>w color<span class="_ _3"></span>,</div><div class="t m0 x4 h6 y89 ff2 fs2 fc0 sc0 ls0 ws0">click<span class="_ _7"> </span>’User<span class="_ _5"> </span>Track<span class="_ _5"> </span>after<span class="_ _7"> </span>submission’,<span class="_ _7"> </span>and<span class="_ _7"> </span>edit<span class="_ _7"> </span>conﬁg<span class="_ _5"> </span>to<span class="_ _7"> </span>’<span class="_ _1"></span>visibility=2<span class="_ _7"> </span>itemRgb=&quot;On&quot;’.<span class="_ _1e"> </span>Color<span class="_ _7"> </span>see</div><div class="t m0 x4 h6 y8a ff2 fs2 fc0 sc0 ls0 ws0">Github page for more example.</div><div class="t m0 x0 h5 y8b ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 y3e ff3 fs2 fc0 sc0 ls0 ws0">icnv_output_to_gb(chr,<span class="_ _6"> </span>icnv.output)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf5" class="pf w0 h0" data-page-no="5"><div class="pc pc5 w0 h0"><img class="bi x0 y8c w1 hb" alt="" src="bg5.png"/><div class="t m0 x0 h6 y1e ff6 fs2 fc0 sc0 ls0 ws0">output_list<span class="_ _1f"> </span><span class="ff2">5</span></div><div class="t m0 x0 h5 y40 ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 y8d ff3 fs2 fc0 sc0 ls0 ws0">chr<span class="_ _c"> </span><span class="ff2">CNV chromosome</span></div><div class="t m0 x4 h6 y8e ff3 fs2 fc0 sc0 ls0 ws0">icnv.output<span class="_ _e"> </span><span class="ff2">output from output_list_function</span></div><div class="t m0 x0 h5 y8f ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1d"></span>alue</div><div class="t m0 x4 h6 y90 ff2 fs2 fc0 sc0 ls0 ws0">matrix for Genome browser</div><div class="t m0 x0 h5 y91 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 y92 ff3 fs3 fc0 sc0 ls0 ws0">icnv.output<span class="_ _9"> </span>=<span class="_ _9"> </span>output_list(icnv_res=icnv_res,sampleid=sampname_qc,<span class="_ _9"> </span>CN=0,<span class="_ _9"> </span>min_size=10000)</div><div class="t m0 x4 h8 y93 ff3 fs3 fc0 sc0 ls0 ws0">gb_input<span class="_ _9"> </span>=<span class="_ _9"> </span>icnv_output_to_gb(chr=22,icnv.output)</div><div class="t m0 x4 h8 y94 ff3 fs3 fc0 sc0 ls0 ws0">write.table(gb_input,file=<span class="ff5">&apos;</span>icnv_res_gb_chr22.tab<span class="ff5">&apos;</span>,quote=F,col.names=F,row.names=F)</div><div class="t m0 x3 h4 y95 ff3 fs2 fc0 sc0 ls0 ws0">output_list<span class="_ _20"> </span><span class="ff4">Generate ouput list.</span></div><div class="t m0 x0 h5 y96 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 y97 ff2 fs2 fc0 sc0 ls0 ws0">Generate human readable output from result calculated by iCNV_detection function</div><div class="t m0 x0 h5 y98 ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 y99 ff3 fs2 fc0 sc0 ls0 ws0">output_list(icnv_res,<span class="_ _6"> </span>sampleid<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>CN<span class="_ _6"> </span>=<span class="_ _b"> </span>0,<span class="_ _6"> </span>min_size<span class="_ _6"> </span>=<span class="_ _6"> </span>0)</div><div class="t m0 x0 h5 y9a ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 y9b ff3 fs2 fc0 sc0 ls0 ws0">sampleid<span class="_ _d"> </span><span class="ff2">the name of the sample, same order as the input</span></div><div class="t m0 x4 h6 y9c ff3 fs2 fc0 sc0 ls0 ws0">CN<span class="_ _11"> </span><span class="ff2">An indicator<span class="_ _4"> </span>v<span class="_ _3"></span>ariable<span class="_ _4"> </span>with v<span class="_ _3"></span>alue<span class="_ _4"> </span>0,1 for whether<span class="_ _4"> </span>exact copy number<span class="_ _4"> </span>inferred in</span></div><div class="t m0 x5 h6 y9d ff2 fs2 fc0 sc0 ls0 ws0">iCNV_detection.<span class="_ _4"> </span>0 no exact CN, 1 exact CN. Def<span class="_ _3"></span>ault 0.</div><div class="t m0 x4 h6 y9e ff3 fs2 fc0 sc0 ls0 ws0">min_size<span class="_ _d"> </span><span class="ff2">A<span class="_ _5"> </span>integer<span class="_ _5"> </span>which<span class="_ _7"> </span>indicate<span class="_ _5"> </span>the<span class="_ _5"> </span>minimum<span class="_ _7"> </span>length<span class="_ _5"> </span>of<span class="_ _5"> </span>the<span class="_ _7"> </span>CNV<span class="_ _5"> </span>you<span class="_ _7"> </span>are<span class="_ _5"> </span>interested</span></div><div class="t m0 x5 h6 y9f ff2 fs2 fc0 sc0 ls0 ws0">in.<span class="_ _7"> </span>This<span class="_ _4"> </span>could<span class="_ _4"> </span>remo<span class="_ _3"></span>ve<span class="_ _4"> </span>super short<span class="_ _4"> </span>CNVs<span class="_ _4"> </span>due to<span class="_ _4"> </span>noise.<span class="_ _7"> </span>Default<span class="_ _4"> </span>0.<span class="_ _7"> </span>Recommend</div><div class="t m0 x5 h6 ya0 ff2 fs2 fc0 sc0 ls0 ws0">1000.</div><div class="t m0 x4 h6 ya1 ff3 fs2 fc0 sc0 ls0 ws0">testres<span class="_ _15"> </span><span class="ff2">CNV inference result.<span class="_ _4"> </span>Output from iCNV_detection()</span></div><div class="t m0 x0 h5 ya2 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1d"></span>alue</div><div class="t m0 x4 h6 ya3 ff2 fs2 fc0 sc0 ls0 ws0">output CNV list of each individual</div><div class="t m0 x0 h5 ya4 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 ya5 ff3 fs3 fc0 sc0 ls0 ws0">icnv.output<span class="_ _9"> </span>=<span class="_ _9"> </span>output_list(icnv_res=icnv_res,sampleid=sampname_qc,<span class="_ _9"> </span>CN=0)</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf6" class="pf w0 h0" data-page-no="6"><div class="pc pc6 w0 h0"><img class="bi x0 ya6 w1 hc" alt="" src="bg6.png"/><div class="t m0 x0 h6 y1e ff2 fs2 fc0 sc0 ls0 ws0">6<span class="_ _21"> </span><span class="ff6">plotHMMscore</span></div><div class="t m0 x3 h4 ya7 ff3 fs2 fc0 sc0 ls0 ws0">plot_intensity<span class="_ _17"> </span><span class="ff4">plot out the NGS plr or array lrr<span class="_ _1d"></span>.</span></div><div class="t m0 x0 h5 ya8 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 ya9 ff2 fs2 fc0 sc0 ls0 ws0">For quality checking purpose during intermediate steps</div><div class="t m0 x0 h5 yaa ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 yab ff3 fs2 fc0 sc0 ls0 ws0">plot_intensity(intensity,<span class="_ _6"> </span>chr)</div><div class="t m0 x0 h5 yac ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 yad ff3 fs2 fc0 sc0 ls0 ws0">intensity<span class="_ _22"> </span><span class="ff2">Specify the ngs_plr object generated by CODEX or SNP array<span class="_ _1"></span>.</span></div><div class="t m0 x4 h6 yae ff3 fs2 fc0 sc0 ls0 ws0">chr<span class="_ _c"> </span><span class="ff2">Specify the chromosome you want to generate.<span class="_ _4"> </span>Must be of int from 1-22</span></div><div class="t m0 x0 h5 yaf ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1d"></span>alue</div><div class="t m0 x4 h6 yb0 ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 yb1 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 yb2 ff3 fs3 fc0 sc0 ls0 ws0">plot_intensity(ngs_plr,chr)</div><div class="t m0 x4 h8 yb3 ff3 fs3 fc0 sc0 ls0 ws0">plot_intensity(snp_lrr,chr)</div><div class="t m0 x3 h4 yb4 ff3 fs2 fc0 sc0 ls0 ws0">plotHMMscore<span class="_ _23"> </span><span class="ff4">Plot CNV infer<span class="_ _3"></span>ence scor<span class="_ _3"></span>e.</span></div><div class="t m0 x0 h5 yb5 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 yb6 ff2 fs2 fc0 sc0 ls0 ws0">Plot out CNV<span class="_ _4"> </span>inference score.<span class="_ _5"> </span>Each row is a<span class="_ _4"> </span>sample, each column<span class="_ _4"> </span>is a SNP or, exon (WES) or bin</div><div class="t m0 x4 h6 yb7 ff2 fs2 fc0 sc0 ls0 ws0">(WGS). Red color indicate score fa<span class="_ _3"></span>vor duplication whereas blue fa<span class="_ _3"></span>vor deletion.</div><div class="t m0 x0 h5 yb8 ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 yb9 ff3 fs2 fc0 sc0 ls0 ws0">plotHMMscore(icnv_res,<span class="_ _6"> </span>h<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL,<span class="_ _6"> </span>t<span class="_ _6"> </span>=<span class="_ _b"> </span>NULL,<span class="_ _6"> </span>subj<span class="_ _6"> </span>=<span class="_ _6"> </span>&quot;score<span class="_ _6"> </span>plot&quot;,</div><div class="t m0 x7 h7 yba ff3 fs2 fc0 sc0 ls0 ws0">output<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL)</div><div class="t m0 x0 h5 ybb ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 ybc ff3 fs2 fc0 sc0 ls0 ws0">icnv_res<span class="_ _d"> </span><span class="ff2">CNV inference result.<span class="_ _4"> </span>Result from iCNV_detection() (i.e.<span class="_ _5"> </span>iCNV_detection(...))</span></div><div class="t m0 x4 h6 ybd ff3 fs2 fc0 sc0 ls0 ws0">h<span class="_ _24"> </span><span class="ff2">start position of this plot.<span class="_ _4"> </span>Default Start of the whole chromosome</span></div><div class="t m0 x4 h6 ybe ff3 fs2 fc0 sc0 ls0 ws0">t<span class="_ _24"> </span><span class="ff2">end position of this plot.<span class="_ _4"> </span>Default End of the whole chromosome</span></div><div class="t m0 x4 h6 ybf ff3 fs2 fc0 sc0 ls0 ws0">output<span class="_ _1b"> </span><span class="ff2">generated<span class="_ _5"> </span>from<span class="_ _5"> </span>output_list_function.<span class="_ _b"> </span>If<span class="_ _5"> </span>it<span class="_ _5"> </span>isn’t<span class="_ _4"> </span>null,<span class="_ _7"> </span>only<span class="_ _4"> </span>CNVs<span class="_ _5"> </span>in<span class="_ _5"> </span>output<span class="_ _5"> </span>ﬁle</span></div><div class="t m0 x5 h6 yc0 ff2 fs2 fc0 sc0 ls0 ws0">will be highlighted.<span class="_ _4"> </span>Default NULL</div><div class="t m0 x4 h6 y3e ff3 fs2 fc0 sc0 ls0 ws0">title<span class="_ _13"> </span><span class="ff2">of this plot.<span class="_ _4"> </span>Default &quot;score plot&quot;</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf7" class="pf w0 h0" data-page-no="7"><div class="pc pc7 w0 h0"><img class="bi x0 yc1 w1 h1" alt="" src="bg7.png"/><div class="t m0 x0 h6 y1e ff6 fs2 fc0 sc0 ls0 ws0">plotindi<span class="_ _25"> </span><span class="ff2">7</span></div><div class="t m0 x0 h5 y40 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1d"></span>alue</div><div class="t m0 x4 h6 yc2 ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 yc3 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 yc4 ff3 fs3 fc0 sc0 ls0 ws0">icnv_res<span class="_ _9"> </span>=<span class="_ _9"> </span>iCNV_detection(...)</div><div class="t m0 x4 h8 yc5 ff3 fs3 fc0 sc0 ls0 ws0">pdf(file=paste0(projname,<span class="ff5">&apos;</span>.pdf<span class="ff5">&apos;</span>),width=13,height<span class="_ _9"> </span>=<span class="_ _9"> </span>10)</div><div class="t m0 x4 h8 yc6 ff3 fs3 fc0 sc0 ls0 ws0">plotHMMscore(icnv_res,h=100000,<span class="_ _9"> </span>t=200000,<span class="_ _9"> </span>subj=<span class="ff5">&apos;</span>my<span class="_ _9"> </span>favorite<span class="_ _9"> </span>subject<span class="ff5">&apos;</span>)</div><div class="t m0 x4 h8 yc7 ff3 fs3 fc0 sc0 ls0 ws0">dev.off()</div><div class="t m0 x3 h4 yc8 ff3 fs2 fc0 sc0 ls0 ws0">plotindi<span class="_ _26"> </span><span class="ff4">Individual sample plot</span></div><div class="t m0 x0 h5 yc9 ff1 fs2 fc0 sc0 ls0 ws0">Description</div><div class="t m0 x4 h6 yca ff2 fs2 fc0 sc0 ls0 ws0">Plot relationship<span class="_ _4"> </span>between platforms and<span class="_ _4"> </span>features for<span class="_ _4"> </span>each individual.<span class="_ _5"> </span>Only work<span class="_ _4"> </span>for muli-platform</div><div class="t m0 x4 h6 ycb ff2 fs2 fc0 sc0 ls0 ws0">inference.</div><div class="t m0 x0 h5 ycc ff1 fs2 fc0 sc0 ls0 ws0">Usage</div><div class="t m0 x4 h7 ycd ff3 fs2 fc0 sc0 ls0 ws0">plotindi(ngs_plr,<span class="_ _6"> </span>snp_lrr,<span class="_ _6"> </span>ngs_baf,<span class="_ _6"> </span>snp_baf,<span class="_ _6"> </span>ngs_plr.pos,<span class="_ _6"> </span>snp_lrr.pos,</div><div class="t m0 x7 h7 yce ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf.pos,<span class="_ _6"> </span>snp_baf.pos,<span class="_ _6"> </span>icnvres,<span class="_ _6"> </span>I,<span class="_ _6"> </span>h<span class="_ _6"> </span>=<span class="_ _b"> </span>NULL,<span class="_ _6"> </span>t<span class="_ _6"> </span>=<span class="_ _6"> </span>NULL)</div><div class="t m0 x0 h5 ycf ff1 fs2 fc0 sc0 ls0 ws0">Arguments</div><div class="t m0 x4 h6 yd0 ff3 fs2 fc0 sc0 ls0 ws0">ngs_plr<span class="_ _15"> </span><span class="ff2">A<span class="_ _4"> </span>list<span class="_ _5"> </span>of<span class="_ _4"> </span>NGS<span class="_ _5"> </span>intensity<span class="_ _4"> </span>data.<span class="_ _6"> </span>Each<span class="_ _5"> </span>entry<span class="_ _4"> </span>is<span class="_ _4"> </span>an<span class="_ _5"> </span>indi<span class="_ _3"></span>vidual.<span class="_ _6"> </span>If<span class="_ _5"> </span>no<span class="_ _4"> </span>NGS<span class="_ _4"> </span>data,<span class="_ _5"> </span>no</span></div><div class="t m0 x5 h6 yd1 ff2 fs2 fc0 sc0 ls0 ws0">need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 yd2 ff3 fs2 fc0 sc0 ls0 ws0">snp_lrr<span class="_ _15"> </span><span class="ff2">A list<span class="_ _4"> </span>of SNP<span class="_ _4"> </span>array intensity<span class="_ _4"> </span>data.<span class="_ _5"> </span>Each<span class="_ _4"> </span>entry is<span class="_ _4"> </span>an individual.<span class="_ _5"> </span>If<span class="_ _4"> </span>no SNP array</span></div><div class="t m0 x5 h6 yd3 ff2 fs2 fc0 sc0 ls0 ws0">data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 yd4 ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf<span class="_ _15"> </span><span class="ff2">A<span class="_ _4"> </span>list of<span class="_ _4"> </span>NGS<span class="_ _4"> </span>B<span class="_ _1"></span>AF<span class="_ _4"> </span>data.<span class="_ _8"> </span>Each<span class="_ _4"> </span>entry is<span class="_ _4"> </span>an<span class="_ _4"> </span>indi<span class="_ _3"></span>vidual.<span class="_ _7"> </span>If<span class="_ _4"> </span>no<span class="_ _4"> </span>NGS data,<span class="_ _4"> </span>no<span class="_ _4"> </span>need</span></div><div class="t m0 x5 h6 yd5 ff2 fs2 fc0 sc0 ls0 ws0">to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 yd6 ff3 fs2 fc0 sc0 ls0 ws0">snp_baf<span class="_ _15"> </span><span class="ff2">A list<span class="_ _2"> </span>of SNP array<span class="_ _2"> </span>BAF<span class="_ _2"> </span>data.<span class="_ _4"> </span>Each entry is an<span class="_ _2"> </span>individual.<span class="_ _4"> </span>If no<span class="_ _2"> </span>SNP array data,</span></div><div class="t m0 x5 h6 yd7 ff2 fs2 fc0 sc0 ls0 ws0">no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 yd8 ff3 fs2 fc0 sc0 ls0 ws0">ngs_plr.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _5"> </span>list<span class="_ _4"> </span>of<span class="_ _5"> </span>NGS<span class="_ _4"> </span>intensity<span class="_ _5"> </span>postion<span class="_ _4"> </span>data.<span class="_ _b"> </span>Each<span class="_ _5"> </span>entry<span class="_ _5"> </span>is<span class="_ _4"> </span>an<span class="_ _5"> </span>indi<span class="_ _3"></span>vidual<span class="_ _5"> </span>with<span class="_ _4"> </span>dimen-</span></div><div class="t m0 x5 h6 yd9 ff2 fs2 fc0 sc0 ls0 ws0">sion= (#of bins<span class="_ _4"> </span>or exons, 2(start and end<span class="_ _4"> </span>position)).<span class="_ _5"> </span>If no NGS data,<span class="_ _4"> </span>no need to</div><div class="t m0 x5 h6 yda ff2 fs2 fc0 sc0 ls0 ws0">specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 ydb ff3 fs2 fc0 sc0 ls0 ws0">snp_lrr.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _7"> </span>list<span class="_ _7"> </span>of<span class="_ _7"> </span>SNP<span class="_ _7"> </span>array<span class="_ _7"> </span>intensity<span class="_ _7"> </span>postion<span class="_ _5"> </span>data.<span class="_ _18"> </span>Each<span class="_ _7"> </span>entry<span class="_ _7"> </span>is<span class="_ _7"> </span>an<span class="_ _7"> </span>indi<span class="_ _3"></span>vidual<span class="_ _7"> </span>with</span></div><div class="t m0 x5 h6 ydc ff2 fs2 fc0 sc0 ls0 ws0">length=#of SNPs.<span class="_ _4"> </span>If no SNP array data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 ydd ff3 fs2 fc0 sc0 ls0 ws0">ngs_baf.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _4"> </span>list<span class="_ _5"> </span>of<span class="_ _5"> </span>NGS<span class="_ _4"> </span>B<span class="_ _3"></span>AF<span class="_ _5"> </span>postion<span class="_ _4"> </span>data.<span class="_ _b"> </span>Each<span class="_ _5"> </span>entry<span class="_ _4"> </span>is<span class="_ _5"> </span>an<span class="_ _4"> </span>individual<span class="_ _4"> </span>with<span class="_ _5"> </span>length=#of</span></div><div class="t m0 x5 h6 yde ff2 fs2 fc0 sc0 ls0 ws0">B<span class="_ _3"></span>AFs.<span class="_ _4"> </span>If no NGS data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 ydf ff3 fs2 fc0 sc0 ls0 ws0">snp_baf.pos<span class="_ _e"> </span><span class="ff2">A<span class="_ _19"> </span>list<span class="_ _2"> </span>of<span class="_ _2"> </span>SNP<span class="_ _19"> </span>array<span class="_ _2"> </span>BAF<span class="_ _19"> </span>postion<span class="_ _19"> </span>data.<span class="_ _4"> </span>Each<span class="_ _2"> </span>entry<span class="_ _19"> </span>is<span class="_ _2"> </span>an<span class="_ _2"> </span>individual<span class="_ _19"> </span>with<span class="_ _19"> </span>length=#of</span></div><div class="t m0 x5 h6 ye0 ff2 fs2 fc0 sc0 ls0 ws0">B<span class="_ _3"></span>AFs.<span class="_ _4"> </span>If no SNP array data, no need to specify<span class="_ _1"></span>.</div><div class="t m0 x4 h6 ye1 ff3 fs2 fc0 sc0 ls0 ws0">icnvres<span class="_ _15"> </span><span class="ff2">CNV inference result.<span class="_ _4"> </span>The output from iCNV_detection()</span></div><div class="t m0 x4 h6 ye2 ff3 fs2 fc0 sc0 ls0 ws0">I<span class="_ _24"> </span><span class="ff2">Indicating the position of the individual to plot</span></div><div class="t m0 x4 h6 ye3 ff3 fs2 fc0 sc0 ls0 ws0">h<span class="_ _24"> </span><span class="ff2">start position of this plot.<span class="_ _4"> </span>Default Start of the whole chromosome</span></div><div class="t m0 x4 h6 y3e ff3 fs2 fc0 sc0 ls0 ws0">t<span class="_ _24"> </span><span class="ff2">end position of this plot.<span class="_ _4"> </span>Default End of the whole chromosome</span></div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf8" class="pf w0 h0" data-page-no="8"><div class="pc pc8 w0 h0"><div class="t m0 x0 h6 y1e ff2 fs2 fc0 sc0 ls0 ws0">8<span class="_ _25"> </span><span class="ff6">plotindi</span></div><div class="t m0 x0 h5 y40 ff1 fs2 fc0 sc0 ls0 ws0">V<span class="_ _1d"></span>alue</div><div class="t m0 x4 h6 ye4 ff2 fs2 fc0 sc0 ls0 ws0">void</div><div class="t m0 x0 h5 ye5 ff1 fs2 fc0 sc0 ls0 ws0">Examples</div><div class="t m0 x4 h8 ye6 ff3 fs3 fc0 sc0 ls0 ws0">pdf(file=paste0(projname,<span class="ff5">&apos;</span>.pdf<span class="ff5">&apos;</span>),width=13,height<span class="_ _9"> </span>=<span class="_ _9"> </span>10)</div><div class="t m0 x4 h8 ye7 ff3 fs3 fc0 sc0 ls0 ws0">plotindi(r1L,r2L,baf1,baf2,rpos1,rpos2,bpos1,bpos2,icnvres,I,h=100000,<span class="_ _9"> </span>t=200000)</div><div class="t m0 x4 h8 ye8 ff3 fs3 fc0 sc0 ls0 ws0">dev.off()</div></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
<div id="pf9" class="pf w0 h0" data-page-no="9"><div class="pc pc9 w0 h0"><div class="t m0 x0 hd ye9 ff1 fs4 fc0 sc0 ls0 ws0">Index</div><div class="t m0 x0 he yea ff7 fs2 fc0 sc0 ls0 ws0">∗<span class="ff2">T<span class="_ _1"></span>opic<span class="_ _4"> </span><span class="ff1 fs1">B<span class="_ _3"></span>AF<span class="_ _1d"></span>,</span></span></div><div class="t m0 x9 h6 yeb ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 he yec ff7 fs2 fc0 sc0 ls0 ws0">∗<span class="ff2">T<span class="_ _1"></span>opic<span class="_ _4"> </span><span class="ff1 fs1">CNV<span class="_ _27"></span>,</span></span></div><div class="t m0 x9 h6 yed ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 he yee ff7 fs2 fc0 sc0 ls0 ws0">∗<span class="ff2">T<span class="_ _1"></span>opic<span class="_ _4"> </span><span class="ff1 fs1">Intensity</span></span></div><div class="t m0 x9 h6 yef ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 he yf0 ff7 fs2 fc0 sc0 ls0 ws0">∗<span class="ff2">T<span class="_ _1"></span>opic<span class="_ _4"> </span><span class="ff1 fs1">Platf<span class="_ _3"></span>orm</span></span></div><div class="t m0 x9 h6 yf1 ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 he yf2 ff7 fs2 fc0 sc0 ls0 ws0">∗<span class="ff2">T<span class="_ _1"></span>opic<span class="_ _4"> </span><span class="ff1 fs1">integration,</span></span></div><div class="t m0 x9 h6 yf3 ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 h6 yf4 ff3 fs2 fc0 sc0 ls0 ws0">bambaf_from_vcf<span class="ff2">, <span class="fc1">1</span></span></div><div class="t m0 x0 h6 yf5 ff3 fs2 fc0 sc0 ls0 ws0">bed_generator<span class="ff2">, <span class="fc1">2</span></span></div><div class="t m0 x0 h6 yf6 ff3 fs2 fc0 sc0 ls0 ws0">get_array_intput<span class="ff2">, <span class="fc1">2</span></span></div><div class="t m0 x0 h6 yf7 ff3 fs2 fc0 sc0 ls0 ws0">iCNV_detection<span class="ff2">, <span class="fc1">3</span></span></div><div class="t m0 x0 h6 yf8 ff3 fs2 fc0 sc0 ls0 ws0">icnv_output_to_gb<span class="ff2">, <span class="fc1">4</span></span></div><div class="t m0 x0 h6 yf9 ff3 fs2 fc0 sc0 ls0 ws0">output_list<span class="ff2">, <span class="fc1">5</span></span></div><div class="t m0 x0 h6 yfa ff3 fs2 fc0 sc0 ls0 ws0">plot_intensity<span class="ff2">, <span class="fc1">6</span></span></div><div class="t m0 x0 h6 yfb ff3 fs2 fc0 sc0 ls0 ws0">plotHMMscore<span class="ff2">, <span class="fc1">6</span></span></div><div class="t m0 x0 h6 yfc ff3 fs2 fc0 sc0 ls0 ws0">plotindi<span class="ff2">, <span class="fc1">7</span></span></div><div class="t m0 x6 h6 y1c ff2 fs2 fc0 sc0 ls0 ws0">9</div><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:193.995000px;bottom:611.140000px;width:6.974000px;height:10.186000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:193.995000px;bottom:587.229000px;width:6.974000px;height:10.187000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:193.995000px;bottom:563.319000px;width:6.974000px;height:10.187000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:193.995000px;bottom:539.409000px;width:6.974000px;height:10.186000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:193.995000px;bottom:515.498000px;width:6.974000px;height:10.187000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf1" data-dest-detail='[1,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:179.051000px;bottom:493.580000px;width:6.974000px;height:10.227000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf2" data-dest-detail='[2,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:169.089000px;bottom:481.331000px;width:6.973000px;height:10.481000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf2" data-dest-detail='[2,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:184.033000px;bottom:459.413000px;width:6.973000px;height:10.481000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf3" data-dest-detail='[3,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:174.070000px;bottom:437.790000px;width:6.974000px;height:10.186000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf4" data-dest-detail='[4,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:189.014000px;bottom:425.540000px;width:6.974000px;height:10.481000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf5" data-dest-detail='[5,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:159.126000px;bottom:403.692000px;width:6.974000px;height:10.411000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf6" data-dest-detail='[6,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:174.070000px;bottom:381.705000px;width:6.974000px;height:10.480000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf6" data-dest-detail='[6,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:164.107000px;bottom:369.819000px;width:6.974000px;height:10.411000px;background-color:rgba(255,255,255,0.000001);"></div></a><a class="l" href="#pf7" data-dest-detail='[7,"XYZ",99.346,773.487,null]'><div class="d m1" style="border-style:none;position:absolute;left:144.182000px;bottom:357.864000px;width:6.974000px;height:10.212000px;background-color:rgba(255,255,255,0.000001);"></div></a></div><div class="pi" data-data='{"ctm":[1.000000,0.000000,0.000000,1.000000,0.000000,0.000000]}'></div></div>
</div>
<div class="loading-indicator">

</div>
</body>
</html>
